spec/gke/hypervisor/gke_spec.rb

```diff
@@ -0,0 +1,178 @@
+require 'fakefs/spec_helpers'
+
+describe Beaker::Gke do
+  let(:hosts) do
+    the_hosts = make_hosts
+    the_hosts
+  end
+
+  let(:options) do
+    {
+      logger: logger
+    }
+  end
+
+  let(:logger) do
+    logger = instance_double('logger')
+    allow(logger).to receive(:debug)
+    allow(logger).to receive(:info)
+    allow(logger).to receive(:warn)
+    allow(logger).to receive(:error)
+    allow(logger).to receive(:notify)
+    logger
+  end
+
+  let(:config) { instance_double('config') }
+
+  let(:context) do
+    instance_double('context',
+                    api_endpoint: 'v1',
+                    ssl_options: {
+                      verify_ssl: 1,
+                      cert_store: true
+                    },
+                    auth_options: {
+                      bearer_token: 'TOKEN_STRING'
+                    })
+  end
+
+  let(:gke) { ::Beaker::Gke.new(hosts, options) }
+
+  def pass_through_initialization
+    allow(ENV).to receive(:fetch).with('KUBECONFIG').and_return('default_value')
+    allow(ENV).to receive(:fetch).with('GOOGLE_APPLICATION_CREDENTIALS').and_return('default_value')
+    allow(config).to receive(:context).and_return(context)
+    allow(Kubeclient::Config).to receive(:read).with(ENV['KUBECONFIG']).and_return(config)
+  end
+
+  before do
+    FakeFS.deactivate!
+  end
+
+  describe ' # initialize' do
+    let(:env_error_message) { 'OS environment variable KUBECONFIG and GOOGLE_APPLICATION_CREDENTIALS must be set' }
+
+    it 'raises error when KUBECONFIG and GOOGLE_APPLICATION_CREDENTIALS ENV variables are not set' do
+      allow(ENV).to receive(:fetch).with('GOOGLE_APPLICATION_CREDENTIALS').and_raise(KeyError)
+      allow(ENV).to receive(:fetch).with('KUBECONFIG').and_raise(KeyError)
+
+      expect { gke }.to raise_error(ArgumentError, env_error_message)
+    end
+
+    it 'raises error when KUBECONFIG ENV variable is not set' do
+      allow(ENV).to receive(:fetch).with('GOOGLE_APPLICATION_CREDENTIALS').and_return('default_value')
+      allow(ENV).to receive(:fetch).with('KUBECONFIG').and_raise(KeyError)
+
+      expect { gke }.to raise_error(ArgumentError, env_error_message)
+    end
+
+    it 'raises error when GOOGLE_APPLICATION_CREDENTIALS ENV variable is not set' do
+      allow(ENV).to receive(:fetch).with('KUBECONFIG').and_return('default_value')
+      allow(ENV).to receive(:fetch).with('GOOGLE_APPLICATION_CREDENTIALS').and_raise(KeyError)
+
+      expect { gke }.to raise_error(ArgumentError, env_error_message)
+    end
+
+    it 'does not raise ArgumentError when KUBECONFIG and GOOGLE_APPLICATION_CREDENTIALS ENV are set' do
+      RSpec::Expectations.configuration.on_potential_false_positives = :nothing
+      allow(ENV).to receive(:fetch).with('GOOGLE_APPLICATION_CREDENTIALS').and_return('default_value')
+      allow(ENV).to receive(:fetch).with('KUBECONFIG').and_return('default_value')
+
+      expect { gke }.not_to raise_error(ArgumentError, env_error_message)
+    end
```

Comment:
This is passing, even though @client  = client(https://github.com/puppetlabs/beaker-gke/blob/master/lib/beaker/hypervisor/gke.rb#L30) will try to create an instance of kubeclient.

I suggest to drop  `expect { gke }.not_to raise_error(ArgumentError, env_error_message)` and test that the instance variables are set correctly, eg: `expect(gke.hosts).to eql(hosts)`